emsymbolizer.py

```diff
@@ -46,7 +61,27 @@ def symbolize_address_dwarf(module, address):
   vma_adjust = get_codesec_offset(module)
   cmd = [LLVM_SYMBOLIZER, '-e', module.filename, f'--adjust-vma={vma_adjust}',
          str(address)]
-  check_call(cmd)
+  out = shared.run_process(cmd, stdout=subprocess.PIPE).stdout.strip()
+  out_lines = out.splitlines()
+  # Source location regex, e.g., /abc/def.c:3:5
+  SOURCE_LOC_RE = re.compile(r'(.+):(\d+):(\d+)$')
+  loc_infos = []
+  # llvm-dwarfdump prints two lines per location. The first line contains a
+  # function name, and the second contains a source location like
+  # '/abc/def.c:3:5'. If the function or source info is not available, it will
+  # be printed as '??', in which case we store None. If the line and column info
+  # is not available, they will be printed as 0, which we store as is.
+  for i in range(0, len(out_lines), 2):
+    func, loc_str = out_lines[i], out_lines[i + 1]
+    m = SOURCE_LOC_RE.match(loc_str)
+    source, line, column = m.group(1), m.group(2), m.group(3)
+    if func == '??':
+      func = None
+    if source == '??':
+      source = None
+    loc_infos.append(LocationInfo(source, line, column, func))
```

Comment:
if we're only going to print each LocationInfo, do we actually need to collect them into a list?

Reply:
That's a good question... I don't think we need to after all. Will fix it.